### PR TITLE
Feature/44820 document check payload and check events

### PR DESCRIPTION
**Pull request by GuyHarwood**
- [x] 📃 document events
- [x] 🔎 add example payload
- [x] 💰 bonus: add utility function to view received check archives as JSON via http endpoint

### CODE REVIEW

**Comment by GuyHarwood** [comment]
@activemq please note the addition of a new utility API endpoint for retrieving a decompressed copy of the archive from the received check table

File: tslib/src/functions/util-received-check-viewer/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const functionName = 'util-received-check-reader'
+
+function finish (start: number, context: Context): void {
+  const end = performance.now()
+  const durationInMilliseconds = end - start
+  const timeStamp = new Date().toISOString()
+  context.log(`${functionName}: ${timeStamp} run complete: ${durationInMilliseconds} ms`)
+}
+
+const schoolPinSampler: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
```
Comment (jon-shipley) [inline tslib/src/functions/util-received-check-viewer/index.ts:16]: Please rename the const to checkRetriever or something.

**Review by jon-shipley — APPROVED** [review]
lgtm

File: docs/messaging/check-submission/readme.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+The submitted check payload is [compressed to UTF-16](https://github.com/pieroxy/lz-string) before sending.  Once received it is hydrated for validation and marking.
+
+An example payload schema can be found [here](./check-payload.json)
+
+## Pupil Events
+
+| Event Name | Description |
+| --------- | ---------------- |
+| AppError | Application error in the pupil's browser |
+| AppHidden | "This event is triggered when the pupil-app changes from being in the foreground to the background, e.g. when switching to another tab/window in the browser, or to another application." |
```
Comment (jon-shipley) [inline docs/messaging/check-submission/readme.md:12]: This file has inconsistent use of quotes around the description.  Best to remove them.

File: docs/messaging/check-submission/readme.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+| CheckStartedAPICallSucceeded | An API call to MTC succeeded. |
+| CheckStartedApiCalled | An API call to MTC has been sent. |
+| CheckSubmissionAPICallSucceeded | An API call to MTC succeeded.  This API returns the pupil results to MTC. |
+| CheckSubmissionAPIFailed | An API to return the pupil results to MTC failed. |
+| CheckSubmissionApiCalled | An API to return the pupil results to MTC has been called. |
+| CheckSubmissionFailed | "The browser was unable to return the payload to MTC, even after several attempts. No more attempts will be made." |
+| CheckSubmissionPending | The browser has sent the data to MTC and is waiting for a response. |
+| PauseRendered | The Pause screen was displayed in the browser. This allows the pupils to rest for a few seconds before the next question. |
+| PupilPrefsAPICallFailed | An API call failed to update the pupil preferences - e.g. font size and/or colour scheme. |
+| PupilPrefsAPICallSucceeded | "AN API call succeeded, this updates the pupil preferences - e.g. font size and/or colour scheme." |
```
Comment (jon-shipley) [inline docs/messaging/check-submission/readme.md:25]: Capitalisation: Should be "An API ..."

**Comment by GuyHarwood** [comment]
@jon-shipley thanks for the feedback, all updated

**Review by jon-shipley — APPROVED** [review]
LGMT

Thanks for the changes!

**Pull request merged**